test/secrets_test.rb

```diff
@@ -90,5 +99,16 @@ def process
       reply.replace({foo: {bar: 1}}.to_json)
       assert_raises(RuntimeError) { process }
     end
+
+    describe "api failures" do
+      before do
+        stub_request(:get, "https://foo.bar/api/v1/namespaces/default/pods").
+          to_return(status: 500)
+      end
+
+      it "raises when api calls fail" do
+        assert_raises(RuntimeError) { process }
```

Comment:
maybe a `e.message.must_include "Something"` so we can separate it from regular errors